ironfish-rust-nodejs/src/multisig.rs

```diff
@@ -353,3 +357,54 @@ impl NativeSigningPackage {
             .collect()
     }
 }
+
+#[napi(namespace = "multisig")]
+pub fn dkg_round1(
+    self_identity: String,
+    min_signers: u16,
+    participant_identities: Vec<String>,
+) -> Result<DkgRound1Packages> {
+    let self_identity =
+        Identity::deserialize_from(&hex_to_vec_bytes(&self_identity).map_err(to_napi_err)?[..])?;
+    let participant_identities = try_deserialize_identities(participant_identities)?;
+
+    if !participant_identities.contains(&self_identity) {
+        return Err(to_napi_err(
+            "participant_identities must include self_identity",
+        ));
+    }
+
+    let max_signers = u16::try_from(participant_identities.len())
+        .map_err(|_| to_napi_err("too many participants"))?;
+
+    if min_signers > max_signers {
+        return Err(to_napi_err(
+            "min_signers exceeds the number of participants",
+        ));
+    }
+
+    let mut rng = thread_rng();
+    let (secret_package, public_package) = dkg::part1(
+        self_identity.to_frost_identifier(),
+        max_signers,
+        min_signers,
+        &mut rng,
+    )
+    .map_err(to_napi_err)?;
```

Comment:
wondering if we should move logic like this into `ironfish-rust` or `ironfish-frost` as we add more validation, checksums, etc.

I'm not sure if having it in `ironfish-rust-nodejs` will make it difficult to use if someone were trying to build on top of it outside of the TypeScript SDK